src/main/java/me/rayzr522/jsonmessage/JSONMessage.java

```diff
@@ -816,116 +887,128 @@ static int getVersion() {
         private MessageEvent onClick;
         private MessageEvent onHover;
-        private ChatColor color;
+        private String color;
+        private String font;
         private String text;
-
+        
         public MessagePart(String text) {
             this.text = text == null ? "null" : text;
         }
-
+        
         /**
          * Converts this {@link MessagePart} into a {@link JsonObject}
          *
          * @return The Minecraft-compatible {@link JsonObject}
          */
         public JsonObject toJSON() {
             Objects.requireNonNull(text);
-
+            
             JsonObject obj = new JsonObject();
             obj.addProperty("text", text);
-
-            if (color != null) {
-                obj.addProperty("color", color.name().toLowerCase());
+            
+            if (color != null && !color.isEmpty()) {
+                obj.addProperty("color", color.toLowerCase());
             }
-
+            
             for (ChatColor style : styles) {
                 obj.addProperty(stylesToNames.get(style), true);
             }
-
+            
             if (onClick != null) {
                 obj.add("clickEvent", onClick.toJSON());
             }
-
+            
             if (onHover != null) {
                 obj.add("hoverEvent", onHover.toJSON());
             }
-
+            
+            if (font != null) {
+                obj.addProperty("font", font);
+            }
+            
             return obj;
-
+            
         }
-
+        
         /**
          * @return This {@link MessagePart} in legacy-style color/formatting codes
          */
         public String toLegacy() {
             StringBuilder output = new StringBuilder();
             if (color != null) {
-                output.append(color.toString());
+                output.append(color);
             }
             styles.stream()
                     .map(ChatColor::toString)
                     .forEach(output::append);
-
+            
             return output.append(text).toString();
         }
-
+        
         /**
          * @return The click event bound
          */
         public MessageEvent getOnClick() {
             return onClick;
         }
-
+        
         /**
          * @param onClick The new click event to bind
          */
         public void setOnClick(MessageEvent onClick) {
             this.onClick = onClick;
         }
-
+        
         /**
          * @return The hover event bound
          */
         public MessageEvent getOnHover() {
             return onHover;
         }
-
+        
         /**
          * @param onHover The new hover event to bind
          */
         public void setOnHover(MessageEvent onHover) {
             this.onHover = onHover;
         }
-
+        
         /**
          * @return The color
          */
-        public ChatColor getColor() {
+        public String getColor() {
```

Comment:
potentially breaking API change!!! I'd recommend we maintain the original method and have it attempt to get a ChatColor enum value based on the color name (so that all existing code which wouldn't be using hex colors anyways will maintain compatibility), but mark it with @Deprecated and have it throw a friendly error if a ChatColor enum constant cant be found by name.

kinda hacky patch but much much much better than scary java NoSuchMethodErrors flying everywhere from old code hahaha

Reply:
I honestly don't see a reason why someone would use this anywhere.
And it could be difficult with the new 1.16 hex colors as those use `ChatColor.of` from the Bungee API and not org.bukkit.ChatColor.

Reply:
it's not about whether or not you can imagine the usecase for it, by the fact that this is a publically accessible method there is inherently a requirement to maintain backwards compatibility because you never know who *might* have used it, whether they should have or not.

same actually goes for setColor as well, there should still be one that takes `ChatColor` as its parameter. again, as I said, my recommended hack for still being able to return `ChatColor` *isn't* perfect, nor does it need to be. the goal is to simply not throw painful `NoSuchMethodError`s bugs that hard-break things, and warn people that they need to fix their code because the method will eventually be removed.

Reply:
The problem with `getColor()` is, that I can't have 2 of the same naming apparently, no matter if one returns ChatColor and the other a String, so we may need to find a different solution there.
It would be easy to f.e. name the String one `getColorString` or `getColorName`, but I think it's better to have it as `getColor` for the future when the old method might get removed...

Reply:
I now renamed the String version to `getColorValue()` and added a (imo weird) approach to the ChatColor one.
It would throw an IllegalStateException when using hex on non 1.16 servers and returns WHITE as default color output.